canvas_sdk/v1/data/message.py

```diff
@@ -0,0 +1,79 @@
+from django.db import models
+
+
+class TransmissionChannel(models.TextChoices):
+    """Transmission channel."""
+
+    MANUAL = "manual", "Manual"
+    TEXT_MESSAGE = "sms", "Text Message"
+    EMAIL = "email", "Email"
+    NOOP = "noop", "No-op"
+
+
+class Message(models.Model):
+    """Message."""
+
+    class Meta:
+        managed = False
+        db_table = "canvas_sdk_data_api_message_001"
+
+    id = models.UUIDField()
+    dbid = models.BigIntegerField(primary_key=True)
+    created = models.DateTimeField()
+    modified = models.DateTimeField()
+    content = models.TextField()
+    sender = models.ForeignKey(
+        "v1.CanvasUser", on_delete=models.DO_NOTHING, related_name="sent_messages", null=True
+    )
+    recipient = models.ForeignKey(
+        "v1.CanvasUser", on_delete=models.DO_NOTHING, related_name="received_messages", null=True
+    )
```

Comment:
So when you create or edit a message you pass the Patient or Staff externally exposable id, but the data module the key is for a Canvas User and you have to filter for the Patient.user and Staff.user ?

Might want to call that out in the docs cause it could be confusing

Reply:
I'm not sure I understand the question here.
There's no externally_exposable_id in Patient or Staff, only the key which is what's used everywhere for the message effect.

(actually there's no externally_exposable_id in the data module at all)

Reply:
Yeah sorry used to talking in FHIR terms. Just want to make sure we call out in the docs that the Message effects are expecting the Staff.key/Patient.key for sender and recipient. However if they use the data module to view the messages, the sender and recipient are CanvasUsers so they need to find the Staff/Patient through the canvas user. 

So just want to make sure we specify this in the docs because it may not be obvious